src/Build/BackEnd/Components/Scheduler/Scheduler.cs

```diff
@@ -1760,7 +1760,25 @@ private void HandleRequestBlockedByNewRequests(SchedulableRequest parentRequest,
 
                         if (affinityMismatch)
                         {
-                            BuildResult result = new BuildResult(request, new InvalidOperationException(ResourceUtilities.FormatResourceStringStripCodeAndKeyword("AffinityConflict", requestAffinity, existingRequestAffinity)));
+                            ErrorUtilities.VerifyThrowInternalError(
+                                _configCache.HasConfiguration(request.ConfigurationId),
+                                "A request should have a configuration if it makes it this far into the scheduled");
+
+                            var config = _configCache[request.ConfigurationId];
+                            var globalProperties = string.Join(
+                                ";",
+                                config.GlobalProperties.ToDictionary().Select(kvp => $"{kvp.Key}={kvp.Value}"));
```

Comment:
This shouldn't be too perf sensitive, but you should be able to avoid one allocation if you use IEnumerable.Enumerate (and EscapedValue for the value).

Reply:
Also, I don't think there is, but this is making me think there should be an overload of string.Join that takes a character, an enumerable, and a function and calls the function on each part of the enumerable then joins them together with the character. It would make this both cleaner and more efficient. What do you think?

Reply:
All good points, but I'd delay these changes until they actually hurt perf.